async-coap-uri/src/escape/unescape_uri.rs

```diff
@@ -77,23 +79,24 @@ impl<'a> From<UnescapeUri<'a>> for Cow<'a, str> {
 impl<'a> FusedIterator for UnescapeUri<'a> {}
 
 impl<'a> UnescapeUri<'a> {
-    /// Returns the location of the first encountered encoding error, if any.
-    pub fn first_error(&self) -> Option<usize> {
+    /// Returns the first encountered encoding error, if any.
+    pub fn first_error(&self) -> Option<DecodingError> {
```

Comment:
Honestly, having the index of the first encoding error is more useful than what the specific error is, IMHO.